src/rax.c

```diff
@@ -119,18 +120,38 @@ static inline int raxStackPush(raxStack *ts, void *ptr) {
             }
             ts->stack = newalloc;
         }
+
+        uint8_t *old = ts->offsets == ts->static_offsets ? NULL : ts->offsets;
+        size_t new_max = ts->maxitems * 2;
+        ts->offsets = rax_realloc(old, new_max);
+
+        if (ts->offsets == NULL) {
+            ts->offsets = (!old) ? ts->static_offsets : old;
+            ts->oom = 1;
+            errno = ENOMEM;
+            return 0;
+        }
+
+        if (old == NULL) memcpy(ts->offsets, ts->static_offsets, ts->maxitems);
```

Comment:
I'm not sure if we need to deal with oom, because they are always dead code(oom handled by `zmalloc_default_oom()`).

Reply:
I now think I see how the issue developed: antirez/rax does not include zmalloc.{c,h}, hence rax.c directly includes necessary defenses against oom. However Redis Rax does not need the oom defenses because it is handled by zmalloc. 

Removing oom defenses from Redis Rax will simplify the code. Should I do that as part of this PR? 

I will also plan to create or help with an updated standalone rax based upon Redis Rax to replace the obsolete and no longer maintained antirez/rax as there is still quite a bit of interest in it.

Reply:
I will remove the oom defenses from Rax. Simplifying the code and removing dead code will make all other changes easier to conceive, code and review.

Reply:
I prefer not to add oom handle, but we also need to see what other optinion, but we can go on without oom, and if we eventually need oom then it's easy to deal with it.

Reply:
I removed oom handling in the latest commit. It passes `make test` in redis/redis as well as `rax-test` in antirez/rax.